segmentation/utils/spatial_analysis.py

```diff
@@ -192,3 +197,62 @@ def calculate_cluster_spatial_enrichment(all_data, dist_mats, fovs=None,
         stats_xr = spatial_analysis_utils.calculate_enrichment_stats(close_num, close_num_rand)
         stats.loc[fovs[i], :, :] = stats_xr.values
     return values, stats
+
+
+def create_neighborhood_matrix(all_data, dist_matrices, fovs=None, distlim=50):
+    """Calculates the number of neighbor phenotypes for each cell
+
+        Args:
+            all_data: data for the all fov
```

Comment:
it'd be great if you could describe the `all_data` argument a bit more.  is it an xarray?  if so, does it have an assumed shape/structure?  those kinds of things